network/p2p/gossip/metrics.go

```diff
@@ -0,0 +1,185 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package gossip
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/ava-labs/avalanchego/ids"
+)
+
+const (
+	ioLabel    = "io"
+	sentIO     = "sent"
+	receivedIO = "received"
+
+	typeLabel  = "type"
+	pushType   = "push"
+	pullType   = "pull"
+	unsentType = "unsent"
+	sentType   = "sent"
+
+	// dropped indicate that the gossipable element was not added to the set due to some reason.
+	// for sent message, we'll use notReceive below.
+	droppedLabel = "dropped"
```

Comment:
I find the naming for these to be extremely confusing because we use "dropped" even for outbound gossip, for which we can't ever actually drop it (and that's why we have the `not_receive` label). Without looking at the code I wouldn't understand how to interpret these metric names.

Reply:
The alternative is to have sentPushLabels and receivedPushLabels defined as a separate counter vector.